### PR TITLE
make bpf_csum_diff a global helper.

### LINKED ISSUE #624: bpf_csum_diff should be a global helper
Currently bpf_csum_diff is implemented as a xdp helper function in netebpfext.sys. This function simply performs arithmetic operations to compute checksum difference and is not networking specific.

### PR DESCRIPTION
**Pull request by shankarseal**
This PR resolves #624 by moving implementation to bpf_csum_diff to execution_context as a global helper function.

### CODE REVIEW

File: libs/execution_context/ebpf_core.c [hunk 1]
```diff
@@ -1235,17 +1255,15 @@ _ebpf_core_get_current_cpu()
...
 {
-    ebpf_result_t retval;
-    uint8_t* value;
-    retval = ebpf_map_find_entry(
-        map, 0, key, sizeof(&value), (uint8_t*)&value, EBPF_MAP_FLAG_HELPER | EPBF_MAP_FIND_FLAG_DELETE);
-    if (retval != EBPF_SUCCESS)
-        return NULL;
-    else
-        return value;
+    return ebpf_csum_diff(from, from_size, to, to_size, seed);
```
Comment (dthaler) [inline libs/execution_context/ebpf_core.c:1266]: Why do we need both _ebpf_core_csum_diff and ebpf_csum_diff?   Why not just make ebpf_core_csum_diff non-static and have just one on the stack?

File: libs/execution_context/ebpf_general_helpers.c [hunk 2]
```diff
@@ -2,7 +2,8 @@
 // SPDX-License-Identifier: MIT
 
 /**
- * @brief Prototypes for general helper functions (aka global functions) implemented by eBPF core Execution Context.
+ * @brief Defines prototype structures for program information for general helper functions (aka global functions)
+ * implemented by eBPF core Execution Context.
```
Comment (dthaler) [inline libs/execution_context/ebpf_general_helpers.c:6]:
```suggestion
 * implemented by the eBPF core execution context.
```

**Review by dthaler — APPROVED** [review]
Ok modulo comments

File: libs/execution_context/ebpf_core.c (hunk 1 above)
Comment (shankarseal) [reply libs/execution_context/ebpf_core.c]: I did that just for uniformity. But that's not a strong technical reason. So keeping a single function that is callable from unit test app.

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**